### PR TITLE
circuits: benches: match: Add `VALID MATCH MPC` benchmarks

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR makes the following changes:
- Refactors  the `commitment_links` module over the Stark curve -- this only required a one-line change
- Adds benchmarks for the `match` MPC and the `VALID MATCH MPC` collaborative proof (constraint generation, prover latency, and verification latency)

### Testing
- Unit and integration tests pass
- CI will remain red until the refactor is complete

### CODE REVIEW

File: circuits/benches/match.rs [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+                // Execute an MPC to generate the constraints
+                let (party0_time, party1_time) = execute_mock_mpc(|fabric| async move {
+                    // Create a witness to the proof
+                    let witness = create_dummy_witness(&fabric);
+
+                    // Create a constraint system to allocate the constraints within
+                    let pc_gens = PedersenGens::default();
+                    let transcript = HashChainTranscript::new(b"test");
+                    let mut prover =
+                        MpcProver::new_with_fabric(fabric.clone(), transcript, pc_gens);
```
Comment (akirillo) [inline circuits/benches/match.rs:136]:
```suggestion
                        MpcProver::new_with_fabric(fabric.clone(), transcript, &pc_gens);
```

Comment (akirillo) [reply circuits/benches/match.rs:136]: here and elsewhere - at least, my IDE is saying this needs to be a reference

**Review by akirillo — APPROVED** [review]
i think there's a small bug here, but generally lgtm, i like the multiple levels of bench granularity

File: circuits/benches/match.rs (hunk 1 above)
Comment (joeykraut) [reply circuits/benches/match.rs:136]: Run `cargo update` and it should work, that was part of [this change](https://github.com/renegade-fi/mpc-bulletproof/pull/8)

**Pull request merged**